src/app/(nextjs_migration)/(guest)/layout.tsx

```diff
@@ -2,23 +2,70 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { v5 as uuidv5 } from "uuid";
 import { ReactNode } from "react";
 import "../../../client/css/index.css";
 import Image from "next/image";
 import MonitorLogo from "../../../client/images/monitor-logo-transparent@2x.webp";
 import MozillaLogo from "../../../client/images/moz-logo-1color-white-rgb-01.svg";
 import { SignInButton } from "../components/client/SignInButton";
 import { getL10n } from "../../functions/server/l10n";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+import { cookies } from "next/headers";
+import { randomUUID } from "crypto";
+import { PageLoadEvent } from "../components/client/PageLoadEvent";
+import { getExperiments } from "../../functions/server/getExperiments";
 
 export type Props = {
   children: ReactNode;
 };
 
-const GuestLayout = (props: Props) => {
+const GuestLayout = async (props: Props) => {
   const l10n = getL10n();
 
+  // If the user is logged in, use UUID derived from FxA UID as Nimbus user ID.
+  const session = await getServerSession(authOptions);
+  const accountId = session?.user?.subscriber?.fxa_uid;
+  let userId;
+
+  if (accountId) {
```

Comment:
Q: Above we had an extra check for `typeof accountId === "string"`. Do we not need that here?

```
if (accountId && typeof accountId === "string") { ... }
```